### PR TITLE
security(forms): passwords no longer stored with sticky form data

### PR DESCRIPTION
**Pull request by mrclay**
Also makes clearer the documentation for the sticky functions.

(Note this is for defense in depth. This is not a site vulnerability)

### CODE REVIEW

**Review by hypeJunction — APPROVED** [review]

**Comment by juho-jaakkola** [comment]
What's the motivation?

On some sites new users need to answer a lot of questions when registering. It easily gets annoying when password fields need to be refilled in case some other required field was missing or had an invalid value.

**Comment by hypeJunction** [comment]
@juho-jaakkola See https://github.com/Elgg/Elgg/issues/9895

File: engine/classes/Elgg/Forms/StickyForms.php [hunk 1]
```diff
@@ -23,7 +23,13 @@ class StickyForms {
 	 * @return void
 	 */
 	public function makeStickyForm($form_name) {
-	
+
+		$banned_keys = [];
+		// TODO make $banned_keys an argument
+		if (in_array($form_name, ['register', 'useradd', 'usersettings'])) {
+			$banned_keys = ['password', 'password1'];
```
Comment (juho-jaakkola) [inline engine/classes/Elgg/Forms/StickyForms.php]: Second one should be `password2`.

**Review by juho-jaakkola — APPROVED** [review]

File: engine/classes/Elgg/Forms/StickyForms.php (hunk 1 above)
Comment (mrclay) [inline engine/classes/Elgg/Forms/StickyForms.php]: Good catch!

**Comment by mrclay** [comment]
> It easily gets annoying when password fields need to be refilled

We'd want to stick the hashed password in session and have register() somehow push it directly into the user password_hash column.

**Pull request merged**